src/features/workflow/modal/DeployModal.tsx

```diff
@@ -47,8 +48,10 @@ const DeployModal: React.FC = () => {
 
   // local state
   const [ dsName, setDsName ] = useState('')
-  const [ dsNameError, setDsNameError ] = useState()
+  const [debouncedDsName] = useDebounce(dsName, 500)
```

Comment:
whenever you're using a debounce, declare the interval at a file-level constant just after the block of import statements:

```ts
const debounce_rate = 500
```

That way if we need to tune it later it's visible & clear